app/controllers/UserController.scala

```diff
@@ -0,0 +1,35 @@
+package controllers
+
+import scala.concurrent.Future
+
+import com.mohiva.play.silhouette.api.actions.SecuredRequest
+import com.mohiva.play.silhouette.api.Silhouette
+import javax.inject._
+import play.api.libs.json._
+import play.api.mvc._
+
+import models.User
+import utils.auth.DefaultEnv
+
+class UserController @Inject() (
+  cc: ControllerComponents,
+  silhouette: Silhouette[DefaultEnv]
+) extends AbstractController(cc) {
+
+  implicit val userWrite = new Writes[User] {
+    def writes(user: User) = Json.obj(
```

Comment:
did you do it this way so you can specify the field names? I think this also works. 

```
implicit val userWrite = Json.writes[User]
```

Reply:
Yeah, and so we have more flexibility in returning custom fields (like maybe return first + last as single name field instead of separate). Probably also don't need to return everything (like google key)?